src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,373 @@
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var calleeMethodSymbol = semanticModel.GetSymbolInfo(calleeMethodInvocationNode, cancellationToken).GetAnySymbol() as IMethodSymbol;
+            if (calleeMethodSymbol == null)
+            {
+                return;
+            }
+
+            var isOrdinaryOrExtensionMethod = calleeMethodSymbol.IsOrdinaryMethod() || calleeMethodSymbol.IsExtensionMethod;
+            if (calleeMethodSymbol.DeclaredAccessibility != Accessibility.Private || !isOrdinaryOrExtensionMethod)
+            {
+                return;
+            }
+
+            var calleeMethodDeclarationSyntaxReferences = calleeMethodSymbol.DeclaringSyntaxReferences;
+            if (calleeMethodDeclarationSyntaxReferences.Length != 1)
+            {
+                return;
+            }
+
+            var calleeMethodDeclarationSyntaxReference = calleeMethodDeclarationSyntaxReferences[0];
+            // For C# it will return MethodDeclarationSyntax,
+            // For VB it will return MethodStatementSyntax, and what it is expecting is MethodBlockSyntax
+            var calleeMethodDeclarationOrStatementNode = await calleeMethodDeclarationSyntaxReference.GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+            var calleeMethodNode = calleeMethodDeclarationOrStatementNode as TMethodDeclarationSyntax
+                ?? calleeMethodDeclarationOrStatementNode.Parent as TMethodDeclarationSyntax;
+
+            if (calleeMethodNode == null)
+            {
+                return;
+            }
+
+            var inlineExpression = GetInlineExpression(calleeMethodNode);
+            if (_syntaxFacts.IsAwaitExpression(inlineExpression))
+            {
+                // This will make sure there is no duplicate 'await'
+                // Example:
+                // Before:
+                // async Task Caller() => await Callee();
+                // async Task Callee() => await Task.CompletedTask;
+                // After:
+                // async Task Caller() => await Task.CompletedTask;
+                // async Task Callee() => await Task.CompletedTask;
+                // The original inline expression in callee will be 'await Task.CompletedTask'
+                // The caller just need 'Task.CompletedTask' without the 'await'
+                inlineExpression = _syntaxFacts.GetExpressionOfAwaitExpression(inlineExpression) as TExpressionSyntax;
+            }
+
+            if (inlineExpression == null)
+            {
+                return;
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var callerDeclarationNode = _syntaxFacts.GetContainingMemberDeclaration(root, calleeMethodInvocationNode.SpanStart)
+                ?? GetEnclosingMethodLikeNode(calleeMethodNode);
```

Comment:
err.. why do you need both of these calls?  surely one should suffice.

Reply:
Because we want to support inline method from location like Property, it is not a method. I am guessing it has some simpler way to do this?

Reply:
we should chat about this.